test/widgets/autocomplete_test.dart

```diff
@@ -0,0 +1,101 @@
+  final store = await TestZulipBinding.instance.globalStore.perAccount(eg.selfAccount.id);
+  store.addUsers([eg.selfUser, eg.otherUser]);
+  store.addUsers(users);
+  final connection = store.connection as FakeApiConnection;
+
+  // prepare message list data
+  final message = eg.dmMessage(from: eg.selfUser, to: [eg.otherUser]);
+  connection.prepare(json: GetMessagesResult(
+    anchor: message.id,
+    foundNewest: true,
+    foundOldest: true,
+    foundAnchor: true,
+    historyLimited: false,
+    messages: [message],
+  ).toJson());
+
+  await tester.pumpWidget(
+    MaterialApp(
+      home: GlobalStoreWidget(
+        child: PerAccountStoreWidget(
+          accountId: eg.selfAccount.id,
+            child: MessageListPage(
+              narrow: DmNarrow(
+                allRecipientIds: [eg.selfUser.userId, eg.otherUser.userId],
+                selfUserId: eg.selfUser.userId,
+              ))))));
+
+  // global store, per-account store, and message list get loaded
+  await tester.pumpAndSettle();
+
+  // (hint text of compose input in a 1:1 DM)
+  final finder = find.widgetWithText(TextField, 'Message @${eg.otherUser.fullName}');
+  check(finder.evaluate()).isNotEmpty();
+  return finder;
+}
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  group('ComposeAutocomplete', () {
+    testWidgets('options appear, disappear, and change correctly', (WidgetTester tester) async {
+      final user1 = eg.user(userId: 1, fullName: 'User One');
+      final user2 = eg.user(userId: 2, fullName: 'User Two');
+      final user3 = eg.user(userId: 3, fullName: 'User Three');
+      final composeInputFinder = await setupToComposeInput(tester, users: [user1, user2, user3]);
+      final store = await TestZulipBinding.instance.globalStore.perAccount(eg.selfAccount.id);
+      // TODO(#226): Remove this extra edit when this bug is fixed.
+      await tester.enterText(composeInputFinder, 'hello @user ');
+      await tester.enterText(composeInputFinder, 'hello @user t');
+      await tester.pumpAndSettle(); // async computation; options appear
+
+      // check "User Two" and "User Three" appear (but not "User One"); tap Three
+      check(tester.widgetList(find.text('User One'))).isEmpty();
+      tester.widget(find.text('User Two'));
+      await tester.tap(find.text('User Three'));
+      await tester.pump();
+
+      // Autocomplete done; options have disappeared
+      check(tester.widget<TextField>(composeInputFinder).controller!.text)
+        .contains(mention(user3, users: store.users));
+      check(tester.widgetList(find.text('User One'))).isEmpty();
+      check(tester.widgetList(find.text('User Two'))).isEmpty();
+      check(tester.widgetList(find.text('User Three'))).isEmpty();
+
+      // TODO(#226): Remove one of these edits when this bug is fixed.
+      await tester.enterText(composeInputFinder, 'hello @user ');
+      await tester.enterText(composeInputFinder, 'hello @user t');
+
```

Comment:
Are these lines intended? It doesn't seem like they're affecting the test.

Reply:
I added these because I realized that the part that comes after wasn't doing quite what I'd intended:

```dart
      // TODO(#226): Remove one of these edits when this bug is fixed.
      await tester.enterText(composeInputFinder, '');
      await tester.enterText(composeInputFinder, ' ');
      check(tester.widgetList(find.text('User One'))).isEmpty();
      check(tester.widgetList(find.text('User Two'))).isEmpty();
      check(tester.widgetList(find.text('User Three'))).isEmpty();
```

I wanted to check that removing an `AutocompleteIntent` (e.g. by changing the text from 'hello @user two' to ' ') causes the options to disappear. If there were a bug and that didn't happen, then without the added lines, the test would have still passed, because the options had disappeared as a result of tapping one of them.

I'll add comments to try to clarify: there are two things that can cause the options to disappear, and we want to check them both:
- Finishing the autocomplete by tapping an option
- Removing the autocomplete intent (like by clearing out the input)